### ISSUE TITLE
[False Positive] $var is immediately overwritten

### ISSUE THREAD

**Issue by j-applese3d** [kalessil/phpinspectionsea#1887]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), v5.0.0.0 |
| Language level | PHP 8.0 |

Current behaviour
--
![image](https://github.com/kalessil/phpinspectionsea/assets/24463871/6e289173-9c37-49cd-adaa-1b07e2cd2643)

Expected behaviour
--
`$ts` is used before it gets reassigned. However, it's not noticing the usage, so complaining that the second assignment is overwriting the first one rendering the first pointless.

Environment
--
PhpStorm 2023.1.1
Build #PS-231.8770.68, built on April 28, 2023
Runtime version: 17.0.6+10-b829.9 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.19.0-43-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 4800M
Cores: 6
Registry:
    run.processes.with.pty=TRUE

Non-Bundled Plugins:
    com.kalessil.phpStorm.phpInspectionsEA (5.0.0.0)

Current Desktop: KDE

Code to reproduce
---
```php
<?php

$x = [
    [
        'test' => 'yesterday',
        'test1' => '2 days ago',
        'test2' => 'Jan 1',
    ],
];
foreach ($x as $approval) {
    if (!empty($approval['test1'])) {
        $ts = strtotime($approval['test1']);
        $doesItMatter = 'no'; // if you remove this line, it won't report an error
        $tsIndexed[$ts]['some key'][] = [
            'what' => "Approved the document",
        ];
    }

    $ts = strtotime($approval['test2']);
    $tsIndexed[$ts]['some key'][] = [
        'what' => "Requested approval",
    ];
}
```

**Comment by j-applese3d** [comment]
Hmm...
Looks like #1642 might be the same thing -- with a simpler (better) example.

https://github.com/kalessil/phpinspectionsea/issues/1642
